### PR TITLE
fix: Encryption key error handling fix

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- Encryption key related error handling fix
- Platform service error handling and logging improvement

## Why

- When encryption key is changed, its related errors are not correctly displayed

## How

-

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, changes are wrt error handling only when there's a change in encryption key

## Related Issues or PRs

- https://github.com/Zipstack/unstract-sdk/pull/87

## Notes on Testing

- Tested by running a tool locally as well running prompts - to check encryption key errors while accessing adapters. Connectors scenario was not explicitly tested though

## Screenshots
image
image
image

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

**Comment by chandrasekharan-zipstack** [comment]
> ## [Quality Gate Failed](https://sonarcloud.io/dashboard?id=Zipstack_unstract&pullRequest=567) **Quality Gate failed**
> Failed conditions  [15.6% Duplication on New Code](https://sonarcloud.io/component_measures?id=Zipstack_unstract&pullRequest=567&metric=new_duplicated_lines_density&view=list) (required ≤ 3%)
> 
> [See analysis details on SonarCloud](https://sonarcloud.io/dashboard?id=Zipstack_unstract&pullRequest=567)

I couldn't clearly tell which 28 lines trigger this Sonar cloud failure, however it could be linked to error handling / logging code that's written for flask based services like `prompt-service`. I can try moving it to `unstract-core`, however I prefer tackling that in a separate PR to keep this simple for review

File: platform-service/src/unstract/platform_service/main.py [hunk 1]
```diff
@@ -500,11 +512,62 @@ def custom_tool_instance() -> Any:
...
+        )
+    app.logger.error(message)
+
+
+@app.errorhandler(HTTPException)
+def handle_http_exception(e: HTTPException) -> Response:
+    """Return JSON instead of HTML for HTTP errors."""
+    log_exceptions(e)
+    if isinstance(e, APIError):
+        return jsonify(e.to_dict()), e.code
```
Comment (Deepak-Kesavan) [inline platform-service/src/unstract/platform_service/main.py:548]: @chandrasekharan-zipstack Looks like a new sonar issue is introduced here.

image

Comment (chandrasekharan-zipstack) [reply platform-service/src/unstract/platform_service/main.py:548]: Apparently flask is takes care of these types with `Response` itself - will update the type hint

File: frontend/src/hooks/useSessionValid.js [hunk 2]
```diff
@@ -133,18 +135,14 @@ function useSessionValid() {
       // Set the session details
       setSessionDetails(getSessionData(userAndOrgDetails));
     } catch (err) {
-      // TODO: Throw popup error message
```
Comment (johnyrahul) [inline frontend/src/hooks/useSessionValid.js:136]: why id we remove this piece of code? @chandrasekharan-zipstack

Comment (chandrasekharan-zipstack) [reply frontend/src/hooks/useSessionValid.js:136]: @johnyrahul I've added it after the condition that checks for the request status of 412. Previously we weren't raising an alert

Comment (chandrasekharan-zipstack) [reply frontend/src/hooks/useSessionValid.js:136]: Thanks for pointing this out @johnyrahul 
@vishnuszipstack @tahierhussain can you validate this behaviour for subscription period expiry? I've reverted the older implementation but please confirm if its valid to check that for all 402 status codes

Comment (vishnuszipstack) [reply frontend/src/hooks/useSessionValid.js:136]: @chandrasekharan-zipstack LGTM. @tahierhussain Please confirm.

**Review by hari-kuriakose — APPROVED** [review]
@chandrasekharan-zipstack LGTM overall.

File: frontend/src/hooks/useSessionValid.js (hunk 2 above)
Comment (chandrasekharan-zipstack) [reply frontend/src/hooks/useSessionValid.js:136]: Discussed with @tahierhussain to move the navigation to trial expiry page here however it didn't work as expected. Leaving the old code as it is with a comment to review the behaviour later

**Review by tahierhussain — APPROVED** [review]

**Review by Deepak-Kesavan — APPROVED** [review]

**Pull request merged**